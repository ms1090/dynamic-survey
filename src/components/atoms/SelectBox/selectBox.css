.select-box select {
  border-radius: 3px;
  border: solid 1px #8c8c8a;
  color: #5d5d5d;
  cursor: pointer;
  display: block;
  height: 40px;
  margin: 10px 0;
  padding: 10px 16px;
  width: 100%;
}