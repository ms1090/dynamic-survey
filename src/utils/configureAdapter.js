import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// configure adapter to use enzyme in unit test cases
configure({ adapter: new Adapter() });
