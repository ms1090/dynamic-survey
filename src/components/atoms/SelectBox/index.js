import React from 'react';
import PropTypes from 'prop-types';
import './selectBox.css';

/**
 * Display a select box as per customization props
 * @param {String} className CSS class name
 * @param {String} id Input element id
 * @param {String} name Input element name
 * @param {String} label Label for input element
 * @param {Array} options Options for select element
 * @return {React.Element} Return select element UI
 */
const SelectBox = ({
  className,
  id,
  name,
  label,
  options,
  ...others
}) => (
  /* eslint-disable jsx-a11y/label-has-for */
  <label className={`select-box ${className}`} htmlFor={id}>
    <span>{label}</span>
    <select
      id={id}
      {...others}
    >
      {options.map(option => (
        <option value={option.value} key={option.value}>
          {option.label}
        </option>
      ))}
    </select>
  </label>
);
// default values for component props
SelectBox.defaultProps = {
  className: '',
  id: '',
  name: '',
  label: '',
  options: [],
};
// prop types checking
SelectBox.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string,
  name: PropTypes.string,
  label: PropTypes.string,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string,
    value: PropTypes.string,
  })),
};
export default SelectBox;
