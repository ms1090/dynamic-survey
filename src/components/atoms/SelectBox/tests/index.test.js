import React from 'react';
import { shallow } from 'enzyme';
import SelectBox from '..';
import toJson from 'enzyme-to-json';
import 'utils/configureAdapter';

describe('SelectBox Component', () => {
  test('should render correctly', () => {
    const options = [
      {
        label: 'test label 1',
        value: 'test value 1',
      },
      {
        label: 'test label 2',
        value: 'test value 2',
      },
    ];
    const wrapper = shallow(<SelectBox
      id="testId"
      className="test-className"
      name="test-name"
      label="test label"
      options={options}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
