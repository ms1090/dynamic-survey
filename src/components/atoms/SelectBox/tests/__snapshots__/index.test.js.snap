// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectBox Component should render correctly 1`] = `
<label
  className="select-box test-className"
  htmlFor="testId"
>
  <span>
    test label
  </span>
  <select
    id="testId"
  >
    <option
      key="test value 1"
      value="test value 1"
    >
      test label 1
    </option>
    <option
      key="test value 2"
      value="test value 2"
    >
      test label 2
    </option>
  </select>
</label>
`;
